keep-ui/app/workflows/workflow-run-with-alert-modal.tsx

```diff
@@ -38,6 +38,29 @@ export default function AlertTriggerModal({
   const [dependencyValues, setDependencyValues] = useState<
     Record<string, string>
   >({});
+  const [workflows, setWorkflows] = useState<any[]>([]);
+  const [fetchError, setFetchError] = useState(false);
+
+  useEffect(() => {
+    // Fetch workflows from the GitHub URL
+    const fetchWorkflows = async () => {
+      try {
+        const response = await fetch(
+          "https://raw.githubusercontent.com/keephq/keep/main/examples/workflows"
+        );
+        if (!response.ok) {
+          throw new Error("Failed to fetch workflows");
+        }
+        const data = await response.json();
+        setWorkflows(data);
+        setFetchError(false);
+      } catch (error) {
+        setFetchError(true);
+      }
+    };
+
+    fetchWorkflows();
+  }, []);
```

Comment:
worth introducing a new hook for this, wdyt?

Reply:
Crafting a captivating hook is essential to engage the audience from the outset. A well-designed hook can pique the interest of the listeners, setting the tone for a successful presentation. In this context, introducing a new hook can be a valuable strategy to revitalize the presentation and make it more compelling

Reply:
Huh?